glue/viewers/table/qt/tests/test_data_viewer.py

```diff
@@ -460,6 +460,46 @@ def test_incompatible_subset():
     assert refresh2.call_count == 0
 
 
+def test_table_incompatible_attribute():
+    """
+    Regression test for a bug where the table viewer generates an
+    uncaught IncompatibleAttribute error if one toggles the
+    visibility of a dataset iff an invalid subset exists. This
+    occurred because disabled layer_artists were still showing
+    as visible.
+    """
+    app = get_qapp()
+    d1 = Data(x=[1,2,3,4],y=[5,6,7,8],label='d1')
+    d2 = Data(a=['a','b','c'],b=['x','y','z'],label='d2')
+    dc = DataCollection([d1,d2])
+    gapp = GlueApplication(dc)
+    viewer = gapp.new_data_viewer(TableViewer)
+    viewer.add_data(d2)
+
+    #This subset should not be shown in the viewer
+    sg1 = dc.new_subset_group('invalid', d1.id['x'] <= 3)
+
+    assert len(viewer.state.layers) == 2
+    assert not viewer.state.layers[1].visible
+    assert viewer.state.layers[0].visible
```

Comment:
```suggestion
    assert len(viewer.layers) == 2
    assert not viewer.layers[1].visible
    assert viewer.layers[0].visible
```
Note that `data_by_row_and_column()`, including your fix, only operates on `viewer.layers` directly, not `viewer.state`, so is this perhaps what you want to test?
Thanks for looking into this!